tests/Feature/TenantAwareJobs/JobIsDeletedWhenTenantCannotBeRetrievedTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+
+namespace Spatie\Multitenancy\Tests\Feature\TenantAwareJobs;
+
+use Illuminate\Contracts\Bus\Dispatcher;
+use Spatie\Multitenancy\Exceptions\NoCurrentTenant;
+use Spatie\Multitenancy\Models\Tenant;
+use Spatie\Multitenancy\Tests\Feature\TenantAwareJobs\TestClasses\NotTenantAwareTestJob;
+use Spatie\Multitenancy\Tests\Feature\TenantAwareJobs\TestClasses\TenantAwareTestJob;
+use Spatie\Multitenancy\Tests\Feature\TenantAwareJobs\TestClasses\TestJob;
+use Spatie\Multitenancy\Tests\TestCase;
+use Spatie\Valuestore\Valuestore;
+
+class JobIsDeletedWhenTenantCannotBeRetrievedTest extends TestCase
+{
+    private Tenant $tenant;
+
+    private Valuestore $valuestore;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        config()->set('multitenancy.queues_are_tenant_aware_by_default', true);
+        config()->set('queue.default', 'sync');
+
+        $this->tenant = factory(Tenant::class)->create();
+
+        $this->valuestore = Valuestore::make($this->tempFile('tenantAware.json'))->flush();
+    }
+
+    /** @test */
+    public function it_will_fail_a_job_when_no_tenant_is_present_and_queues_are_tenant_aware_by_default()
+    {
+        config()->set('multitenancy.queues_are_tenant_aware_by_default', true);
+
+        $job = new TestJob($this->valuestore);
+
+        try {
+            app(Dispatcher::class)->dispatch($job);
+        } catch (NoCurrentTenant $e) {
```

Comment:
In most cases, I don't use abbreviations. Please use `$exception` instead of `$e`